subprojects/docs/src/docs/userguide/custom_gradle_types.adoc

```diff
@@ -102,6 +102,39 @@ In addition, the property must have one of the following types:
 - `MapProperty`
 - `ConfigurableFileCollection`
 
+[[sec:managed_nested_properties]]
+==== Read-only managed nested properties
+
+To declare a read-only managed nested property, add an abstract getter method for the property to the type annotated with `@link:{javadocPath}/org/gradle/api/tasks/Nested.html[Nested]`.
+The property should not have any setter methods.
+This pattern is useful the if current type has a nested complex type which has the same lifecycle.
+If the lifecycle is different, consider using `Property<NestedType>` instead.
+
+Here is an example of a task type with a `hostAndPath` property:
+
+.Read-only managed nested property
+====
+[source.multi-language-sample,java]
+.UrlProcess.java
+----
+include::{samplesPath}/userguide/plugins/readOnlyNestedProperty/java/buildSrc/src/main/java/UrlProcess.java[tags=url-process]
+include::{samplesPath}/userguide/plugins/readOnlyNestedProperty/java/buildSrc/src/main/java/HostAndPath.java[tags=host-and-path]
+----
+[source.multi-language-sample,kotlin]
+.UrlProcess.kt
+----
+include::{samplesPath}/userguide/plugins/readOnlyNestedProperty/kotlin/buildSrc/src/main/kotlin/UrlProcess.kt[tags=url-process]
+----
+[source.multi-language-sample,groovy]
+.UrlProcess.groovy
+----
+include::{samplesPath}/userguide/plugins/readOnlyNestedProperty/groovy/buildSrc/src/main/groovy/UrlProcess.groovy[tags=url-process]
+----
+====
+
+Note that for a property to be considered a read only managed nested property, _all_ of the property's getter methods must be `public` or `protected` and `abstract` and the property must not have any setter methods.
```

Comment:
Should we be talking about property _type_ here?

Reply:
This is about the property. I could add another sentence about requirements on the property type, though are are no requirements in particular.